pkg/okteto/kubetoken/kubetoken_client.go

```diff
@@ -11,42 +11,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package okteto
+package kubetoken
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
 
+	authenticationv1 "k8s.io/api/authentication/v1"
+
 	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/okteto"
 )
 
 const kubetokenPath = "auth/kubetoken"
 
-type KubeTokenClient struct {
+type cacheSetter interface {
+	Set(contextName, namespace string, token authenticationv1.TokenRequest)
+}
+
+type Client struct {
 	httpClient  *http.Client
 	url         string
 	contextName string
+	namespace   string
+	cache       cacheSetter
 }
 
-func NewKubeTokenClient(contextName, token, namespace string) (*KubeTokenClient, error) {
+func NewClient(contextName, token, namespace, certificate string, isInsecure bool, cache cacheSetter) (*Client, error) {
 	if contextName == "" {
 		return nil, oktetoErrors.ErrCtxNotSet
 	}
 
-	httpClient, url, err := newOktetoHttpClient(contextName, token, fmt.Sprintf("%s/%s", kubetokenPath, namespace))
+	httpClient, url, err := okteto.NewOktetoHttpClient(contextName, token, fmt.Sprintf("%s/%s", kubetokenPath, namespace), certificate, isInsecure)
 	if err != nil {
 		return nil, err
 	}
 
-	return &KubeTokenClient{
+	return &Client{
 		httpClient:  httpClient,
 		url:         url,
 		contextName: contextName,
+		namespace:   namespace,
+		cache:       cache,
 	}, nil
 }
 
-func (c *KubeTokenClient) GetKubeToken() (string, error) {
+func (c *Client) GetKubeToken() (string, error) {
```

Comment:
Maybe we should have this under the OktetoClient since it's related and we don't export another client for just one function. WDYT?

Reply:
Personally I prefer it as it is since we get a more defined boundary on dependencies. I'm flexible to discuss :D